### PR TITLE
Add hook example to access `metadata` 

### LINKED ISSUE #2676: Document datasets' metadata attribute in catalog definition
## Introduction

As of #2440 users are now able to define metadata for their datasets within the catalog. We should add a section to the hooks documentation that provides an example of how to consume this information through hooks. This should be added to this page: https://docs.kedro.org/en/stable/hooks/common_use_cases.html

Related: https://github.com/kedro-org/kedro-plugins/pull/189#issuecomment-1523740768

### PR DESCRIPTION
**Pull request by noklam**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

fix #2676

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by noklam** [comment]
I will keep the PR simple and I created the follow up in #2999

File: docs/source/hooks/common_use_cases.md [hunk 1]
```diff
@@ -200,3 +200,22 @@ HOOKS = (AzureSecretsHook(),)
...
+class MetadataHook:
+    @hook_impl
+    def after_catalog_created(
+        self,
+        catalog: DataCatalog,
+        conf_catalog: Dict[str, Any],
+        conf_creds: Dict[str, Any],
+        feed_dict: Dict[str, Any],
+        save_version: str,
+        load_versions: Dict[str, str],
```
Comment (astrojuanlu) [inline docs/source/hooks/common_use_cases.md:217]: All these parameters are not used right? Maybe we can remove them from the function. Pluggy allows this

File: docs/source/hooks/common_use_cases.md [hunk 2]
```diff
@@ -200,3 +200,22 @@ HOOKS = (AzureSecretsHook(),)
 ```{note}
 Note: `DefaultAzureCredential()` is Azure's recommended approach to authorise access to data in your storage accounts. For more information, consult the [documentation about how to authenticate to Azure and authorize access to blob data](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-python).
 ```
+
+## Use Hook to read `metadata` from `DataCatalog`
```
Comment (stichbury) [inline docs/source/hooks/common_use_cases.md:204]:
```suggestion
## Use a Hook to read `metadata` from `DataCatalog`
```

File: docs/source/hooks/common_use_cases.md [hunk 3]
```diff
@@ -200,3 +200,22 @@ HOOKS = (AzureSecretsHook(),)
...
+We recommend to use the `after_catalog_created` hook if you need to access `metadata` to extend Kedro.
```
Comment (stichbury) [inline docs/source/hooks/common_use_cases.md:205]:
```suggestion
Use the `after_catalog_created` Hook to access `metadata` to extend Kedro.
```

**Review by stichbury — APPROVED** [review]
Approved with minor changes. What's Vale saying about this page?

File: docs/source/hooks/common_use_cases.md (hunk 1 above)
Comment (noklam) [reply docs/source/hooks/common_use_cases.md:217]: Fair point. We do this inconsistently, sometimes we do include signature that wasn't used (I was asked to add that before, can't find it exactly which PR it is). 

For example, `after_node_run` has a redundant `node` argument.
https://docs.kedro.org/en/stable/hooks/examples.html#add-observability-to-your-pipeline

Comment (noklam) [reply docs/source/hooks/common_use_cases.md:217]: @stichbury do you think we should keep all our example minimals? We can do a clean up if this is agreed.

In this case
```
   def after_catalog_created(
        self,
        catalog: DataCatalog,
        conf_catalog: Dict[str, Any],
        conf_creds: Dict[str, Any],
        feed_dict: Dict[str, Any],
        save_version: str,
        load_versions: Dict[str, str])
```

or 
```
   def after_catalog_created(
        self,
        catalog: DataCatalog)
```

will both work.

**Review by astrojuanlu — COMMENTED** [review]
I like this example 👍🏽 but I'm wondering, is this a full solution for https://github.com/kedro-org/kedro/issues/2676? Wondering if we should have some prose somewhere else in the data catalog documentation, otherwise I fear it might be a bit difficult to find. Essentially what @stichbury wrote in https://github.com/kedro-org/kedro/issues/2940 as a response to the reviews on https://github.com/kedro-org/kedro/pull/2888.

What do you think?

**Comment by noklam** [comment]
@astrojuanlu Maybe we split the ticket too much, that should  be https://github.com/kedro-org/kedro/issues/2689 ?

**Review by astrojuanlu — APPROVED** [review]
Too many tickets 🥲 Thanks @noklam !

**Pull request merged**